### PR TITLE
[FEATURE] Add JavaScript Modal API

### PR DESCRIPTION
**Pull request by susannemoog**
See Core Feature 90168 
See Core Feature 63729
See #845

### CODE REVIEW

File: Documentation/ApiOverview/JavaScript/Modules/Index.rst [hunk 1]
```diff
@@ -10,11 +10,144 @@ Various JavaScript Modules
 The following APIs are usually used in the TYPO3 backend by the core itself but may also
 be used by extensions.
 
+Modals
+======
+
+Actions that require a users' attention must be visualized by modal windows.
```
Comment (andreasfernandez) [inline Documentation/ApiOverview/JavaScript/Modules/Index.rst:16]: afaik it must be "require a user's".

File: Documentation/ApiOverview/JavaScript/Modules/Index.rst [hunk 2]
```diff
@@ -10,11 +10,144 @@ Various JavaScript Modules
...
+
+TYPO3 provides an API as basis to create modal windows with severity representation. For better UX,
+if actions (buttons) are attached to the modal, one button must be a positive action. This button
```
Comment (andreasfernandez) [inline Documentation/ApiOverview/JavaScript/Modules/Index.rst:19]: I don't get the last sentence.

File: Documentation/ApiOverview/JavaScript/Modules/Index.rst [hunk 3]
```diff
@@ -10,11 +10,144 @@ Various JavaScript Modules
...
+================== =============== ============ ========================================================================================================
+Name               DataType        Mandatory    Description
+================== =============== ============ ========================================================================================================
+text               string          Yes          The text rendered into the button.
+trigger / action   function        Yes          Callback that's triggered on button click - either simple function or `DeferredAction`/`ImmediateAction`
+active             bool                         Marks the button as active. If true, the button gets the focus.
+btnClass           string                       The css class for the button
+================== =============== ============ ========================================================================================================
+
+Data-Attributes
```
Comment (andreasfernandez) [inline Documentation/ApiOverview/JavaScript/Modules/Index.rst:57]: superfluous dash

File: Documentation/ApiOverview/JavaScript/Modules/Index.rst [hunk 4]
```diff
@@ -10,11 +10,144 @@ Various JavaScript Modules
...
+
+It is also possible to use data-attributes to trigger a modal.
+e.g. on an anchor element, which prevents the default behavior.
+
+========================= ==================================================================
+Name                      Description
+========================= ==================================================================
+data-title                the title text for the modal
+data-content              the content text for the modal
+data-severity             the severity for the modal, default is info (see TYPO3.Severity.*)
```
Comment (andreasfernandez) [inline Documentation/ApiOverview/JavaScript/Modules/Index.rst:68]: :code:\`TYPO3.Severity.*\`

File: Documentation/ApiOverview/JavaScript/Modules/Index.rst [hunk 5]
```diff
@@ -10,11 +10,144 @@ Various JavaScript Modules
...
+Modals
+======
+
+Actions that require a users' attention must be visualized by modal windows.
+
+TYPO3 provides an API as basis to create modal windows with severity representation. For better UX,
+if actions (buttons) are attached to the modal, one button must be a positive action. This button
+should get a `btnClass` and is set as active.
+
+Modals should be used rarely and only for confirmations. For information the TYPO3.Flashmessage API should be used.
```
Comment (andreasfernandez) [inline Documentation/ApiOverview/JavaScript/Modules/Index.rst:22]: :code:\`TYPO3.Flashmessage\`

File: Documentation/ApiOverview/JavaScript/Modules/Index.rst [hunk 6]
```diff
@@ -10,11 +10,144 @@ Various JavaScript Modules
...
+
+Modal Settings
+~~~~~~~~~~~~~~
+
+========= =============== ============ ======================================================================================================
+Name      DataType        Mandatory    Description
+========= =============== ============ ======================================================================================================
+title     string          Yes          The title displayed in the modal
+content   string|jQuery   Yes          The content displayed in the modal
+severity  int                          Represents the severity of a modal. Please see TYPO3.Severity. Default is :code:`TYPO3.Severity.info`.
```
Comment (andreasfernandez) [inline Documentation/ApiOverview/JavaScript/Modules/Index.rst:41]: :code:\`TYPO3.Severity\`

**Review by andreasfernandez — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/JavaScript/Modules/Index.rst [hunk 7]
```diff
@@ -10,11 +10,144 @@ Various JavaScript Modules
...
+data-button-close-text    button text for the close/cancel button
+data-button-ok-text       button text for the ok button
+========================= ==========================================================================
+
+:code:`class="t3js-modal-trigger"` marks the element as modal trigger
+
+Examples
+--------
+
+A basic modal without any specials can be created this way:
```
Comment (flicstar) [inline Documentation/ApiOverview/JavaScript/Modules/Index.rst:79]:
```suggestion
A basic modal (without anything special) can be created this way:
```
OR A basic modal can be created this way:

**Review by andreasfernandez — APPROVED** [review]

**Pull request merged**